### PR TITLE
feat(bitbucket): add discovery support for bitbucket (server & cloud)

### PR DESCRIPTION
**Pull request by schultzp2020**
## Description

Please explain the changes you made here.

## Which issue(s) does this PR fix

- Fixes https://github.com/janus-idp/backstage-showcase/issues/618
- https://issues.redhat.com/browse/RHIDP-905

## PR acceptance criteria

Please make sure that the following steps are complete:

- [ ] GitHub Actions are completed and successful
- [ ] Unit Tests are updated and passing
- [ ] E2E Tests are updated and passing
- [ ] Documentation is updated if necessary (requirement for new features)
- [ ] Add a screenshot if the change is UX/UI related

### CODE REVIEW

File: dynamic-plugins.default.yaml [hunk 1]
```diff
@@ -479,6 +479,49 @@ plugins:
...
+      #           frequency: { minutes: 30 }
+      #           # supports ISO duration, "human duration" as used in code
+      #           timeout: { minutes: 3 }
+      #         workspace: workspace-name
+
+  - package: ./dynamic-plugins/dist/backstage-plugin-catalog-backend-module-bitbucket-server
+    disabled: true
+    pluginConfig:
+      enabled:
+        bitbucketCloud: true
```
Comment (kadel) [inline dynamic-plugins.default.yaml:508]: shouldn't this be `bitbucketServer` or something else then `bitbucketCloud`?

Comment (schultzp2020) [reply dynamic-plugins.default.yaml]: fixed, before we GA we will need to remove the `enabled.xxx` keys anyways.

File: dynamic-plugins.default.yaml [hunk 2]
```diff
@@ -479,6 +479,46 @@ plugins:
...
+      #         catalogPath: /catalog-info.yaml # default value
+      #         filters: # optional
+      #           projectKey: '^apis-.*$' # optional; RegExp
+      #           repoSlug: '^service-.*$' # optional; RegExp
+      #         schedule: # optional; same options as in TaskScheduleDefinition
+      #           # supports cron, ISO duration, "human duration" as used in code
+      #           frequency: { minutes: 30 }
+      #           # supports ISO duration, "human duration" as used in code
+      #           timeout: { minutes: 3 }
+      #         workspace: workspace-name
```
Comment (tumido) [inline dynamic-plugins.default.yaml:500]: Why is this commented out? Also the plugin is disabled by `disabled: true`, this config doesn't get applied unless the plugin is enabled and uncommenting this section doesn't enable the plugin.

File: dynamic-plugins.default.yaml [hunk 3]
```diff
@@ -479,6 +479,46 @@ plugins:
...
+
+  - package: ./dynamic-plugins/dist/backstage-plugin-catalog-backend-module-bitbucket-server
+    disabled: true
+    pluginConfig:
+      # Uncomment the following to enable the Bitbucket Server catalog
```
Comment (tumido) [inline dynamic-plugins.default.yaml:505]: Same as above.

File: dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-bitbucket-cloud/package.json [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    "clean": "backstage-cli package clean",
+    "clean-dynamic-sources": "yarn clean && rm -Rf node_modules",
+    "export-dynamic": "janus-cli package export-dynamic-plugin --embed-package @backstage/plugin-catalog-backend-module-bitbucket-cloud"
+  },
+  "dependencies": {
+    "@backstage/backend-plugin-manager": "npm:@janus-idp/backend-plugin-manager@0.0.2-janus.5",
+    "@backstage/plugin-catalog-backend-module-bitbucket-cloud": "0.1.21"
+  },
+  "devDependencies": {
+    "@janus-idp/cli": "1.3.0"
```
Comment (tumido) [inline dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-bitbucket-cloud/package.json:33]: Why is there a different version than in other wrappers?

File: dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-bitbucket-server/package.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    "clean": "backstage-cli package clean",
+    "clean-dynamic-sources": "yarn clean && rm -Rf node_modules",
+    "export-dynamic": "janus-cli package export-dynamic-plugin --embed-package @backstage/plugin-catalog-backend-module-bitbucket-server"
+  },
+  "dependencies": {
+    "@backstage/backend-plugin-manager": "npm:@janus-idp/backend-plugin-manager@0.0.2-janus.5",
+    "@backstage/plugin-catalog-backend-module-bitbucket-server": "0.1.19"
+  },
+  "devDependencies": {
+    "@janus-idp/cli": "1.3.0"
```
Comment (tumido) [inline dynamic-plugins/wrappers/backstage-plugin-catalog-backend-module-bitbucket-server/package.json:33]: same as above

**Review by tumido — APPROVED** [review]
/lgtm
/approve

File: dynamic-plugins.default.yaml [hunk 6]
```diff
@@ -518,6 +518,44 @@ plugins:
...
 
+  - package: ./dynamic-plugins/dist/backstage-plugin-catalog-backend-module-bitbucket-cloud
+    disabled: true
+    pluginConfig:
+      # https://backstage.io/docs/integrations/bitbucketCloud/discovery#configuration
+      catalog:
+        providers:
+          bitbucketCloud:
+            yourProviderId: # identifies your ingested dataset
+              catalogPath: /catalog-info.yaml # default value
```
Comment (davidfestal) [inline dynamic-plugins.default.yaml:529]: Can we use env variables here and below, in order to have this default config be something meaninful and valid (provided we defined the expected env variables).
This is how it has been until now for other plugins, and also historically for the showcase in general in the `app-config.yaml`.

**Review by davidfestal — CHANGES_REQUESTED** [review]

**Comment by tumido** [comment]
Removing `cherry-pick-1.0.x` label based on a discussion with Brad. This is not to be included in 1.0.

**Review by kadel — APPROVED** [review]

File: dynamic-plugins.default.yaml (hunk 6 above)
Comment (schultzp2020) [reply dynamic-plugins.default.yaml:557]: This was the config I copied from the backstage documentation.

Comment (davidfestal) [reply dynamic-plugins.default.yaml:557]: I get it, but the config fragments in this file  are expected to provide default configs that work.

**Comment by schultzp2020** [comment]
@kadel @davidfestal PTAL?

File: dynamic-plugins.default.yaml (hunk 6 above)
Comment (davidfestal) [reply dynamic-plugins.default.yaml]: Don't we want to fix this ?

File: dynamic-plugins.default.yaml [hunk 7]
```diff
@@ -546,6 +546,26 @@ plugins:
...
               clientSecret: "${KEYCLOAK_CLIENT_SECRET}"
 
+  - package: ./dynamic-plugins/dist/backstage-plugin-catalog-backend-module-bitbucket-cloud
+    disabled: true
+    pluginConfig:
+      # https://backstage.io/docs/integrations/bitbucketCloud/discovery#configuration
+      catalog:
+        providers:
+          bitbucketCloud:
+            yourProviderId: # identifies your ingested dataset
```
Comment (davidfestal) [inline dynamic-plugins.default.yaml:556]: do we really want to call the `yourProviderId`, or use `default`, as for `keycloak`: https://github.com/janus-idp/backstage-showcase/pull/681/files#diff-2a8ec8751c51963220c18905adc57407d70bbb6647e6fe394cc224291b93a11fR542

Comment (schultzp2020) [reply dynamic-plugins.default.yaml:556]: I think `default` should fine fine.

**Review by davidfestal — APPROVED** [review]

**Pull request merged**